src/v/cluster/controller.cc

```diff
@@ -298,7 +298,8 @@ ss::future<> controller::start() {
             config::shard_local_cfg()
               .storage_space_alert_free_threshold_bytes.bind(),
             config::shard_local_cfg()
-              .storage_space_alert_free_threshold_percent.bind());
+              .storage_space_alert_free_threshold_percent.bind(),
```

Comment:
Strange, but not dangerous, I think. Maybe you don't like the alerts / log messages (perhaps because you're monitoring redpandas disk stats directly) but you want to set a hard stop? We could warn easily, that might be a nice enhancement. I'm not sure how to do cross-parameter validation of config at load time though.

Reply:
I think you could do it in the health monitors constructor, and either log some message, or just do something like:
```
storage_min_free_bytes = std::min(storage_min_free_bytes, other_var);
```
so its always less or equal to the other threshold

Reply:
Humm. A slight complication is the fact that you can disable either the _percent or _bytes variant of the alert threshold, and we allow disabling both of them (zero is min value of the config param). So it gets a little more complex, like `min(min_free_bytes, to_bytes(percent_alert_threshold, total), bytes_alert_threshold)`, and you need to enforce a minimum alert threshold as well.

Reply:
I'm also a little hesitant to mess with the configured min_free_bytes threshold, since it is a critical setting.

Reply:
Oh.. also these are dynamic config properties so we'd need to trap (register callback) on any change and enforce invariants and/or log. My preference would be to leave as is, maybe do a followup PR, but let me know what you want. I do like how simple the definitions (see table in Cover Letter) are right now. Pretty easy to communicate and reason about.